src/test/java/mate/academy/bookstore/service/BookServiceImplTest.java

```diff
@@ -0,0 +1,216 @@
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    void save_ShouldSaveBook() {
+        Book book = new Book();
+        book.setId(1L);
+
+        when(bookMapper.toEntity(any(CreateBookRequestDto.class))).thenReturn(book);
+        when(bookRepository.save(any(Book.class))).thenReturn(book);
+        when(bookMapper.toDto(any(Book.class))).thenReturn(new BookDto().setId(1L));
+
+        CreateBookRequestDto requestDto = new CreateBookRequestDto()
+                .setTitle("New Book")
+                .setAuthor("New Author")
+                .setIsbn("3333333333333")
+                .setPrice(new BigDecimal("12.9900"))
+                .setDescription("Description")
+                .setCoverImage("cover.jpg")
+                .setCategoryIds(Set.of(1L));
+
+        BookDto result = bookService.save(requestDto);
+
+        assertNotNull(result);
+        assertEquals(1L, result.getId());
+        verify(bookRepository, times(1)).save(book);
+    }
+
+    @Test
+    void findAll_ShouldReturnAllBooks() {
+        Pageable pageable = mock(Pageable.class);
+        Book book = new Book();
+        book.setId(1L);
+        Page<Book> bookPage = new PageImpl<>(List.of(book));
+
+        when(bookRepository.findAll(pageable)).thenReturn(bookPage);
+        when(bookMapper.toDto(any(Book.class))).thenReturn(new BookDto().setId(1L));
+
+        List<BookDto> result = bookService.findAll(pageable);
+
+        assertNotNull(result);
+        assertEquals(1, result.size());
+        assertEquals(1L, result.get(0).getId());
+    }
+
+    @Test
+    void findById_ShouldReturnBook() {
+        Book book = new Book();
+        book.setId(1L);
+
+        when(bookRepository.findById(anyLong())).thenReturn(Optional.of(book));
+        when(bookMapper.toDto(any(Book.class))).thenReturn(new BookDto().setId(1L));
+
+        BookDto result = bookService.findById(1L);
+
+        assertNotNull(result);
+        assertEquals(1L, result.getId());
+    }
+
+    @Test
+    void findById_ShouldThrowException_WhenBookNotFound() {
+        when(bookRepository.findById(anyLong())).thenReturn(Optional.empty());
+
+        EntityNotFoundException exception = assertThrows(EntityNotFoundException.class,
+                () -> bookService.findById(1L));
+
+        assertEquals("Can't find book by id 1", exception.getMessage());
+    }
+
+    @Test
+    void deleteById_ShouldDeleteBook() {
+        bookService.deleteById(1L);
+        verify(bookRepository, times(1)).deleteById(1L);
+    }
+
+    @Test
+    void update_ShouldUpdateBook() {
+        final BookDto bookDto = new BookDto()
+                .setId(1L)
+                .setTitle("Updated Book")
+                .setAuthor("Updated Author")
+                .setIsbn("3333333333333")
+                .setPrice(new BigDecimal("15.9900"))
+                .setDescription("Updated Description")
+                .setCoverImage("updated.jpg");
```

Comment:
if you have the same entity, you can create one time and reuse this pattern

Reply:
didn't understand about the coverage, because I have 100 percent for BookController, CategoryController, BookService, and CategoryService classes as written in the assignment